mypy/join.py

```diff
@@ -57,12 +59,12 @@ def join_types(s: Type, t: Type) -> Type:
     if isinstance(s, AnyType):
         return s
 
-    if isinstance(s, NoneTyp) and not isinstance(t, Void):
-        return t
-
```

Comment:
But here I suspect we want to keep a version of this with `UninhabitedType`. After all consider what happens when `t` is `UninhabitedType` instead.